File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int size = brown + yellow;
        
        // x: 가로, y: 세로
        int x = 0;
        int y = 0;
        
        // y는 적어도 2보다 커야 yellow가 들어갈 공간이 생김
        for(int i = 3; i < size; i++) {
            y = i;
            if(size % y == 0) {
                x = size / y;
                
                int yellowSize = (x - 2) * (y - 2);
                if(yellowSize == yellow) {
                    break;
                }
            }
        }
        
        int[] answer = {x, y};
        return answer;
    }
}

File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    # 가로: n, 세로: m
    # n >= m
    answer = []
    n, m = 0, 0
    
    # T : 전체 카펫크기
    T = brown + yellow
    for y_m in range(1, yellow + 1):
        y_n = yellow // y_m
        if(y_n * y_m != yellow):
            continue
        
        if y_n < y_m:
            break
        
        # 테두리 한줄은 갈색으로 칠해져 있음
        n = y_n + 2
        m = y_m + 2
        
        if n * m == T:
            answer = [n, m]
            break
    
    return answer